modifiers/mod_ts_prop.cpp

```diff
@@ -0,0 +1,103 @@
+/*********************                                                  */
+/*! \file mod_ts_prop.cpp
+** \verbatim
+** Top contributors (to current version):
+**   Makai Mann, Ahmed Irfan
+** This file is part of the pono project.
+** Copyright (c) 2019 by the authors listed in the file AUTHORS
+** in the top-level source directory) and their institutional affiliations.
+** All rights reserved.  See the file LICENSE in the top-level source
+** directory for licensing information.\endverbatim
+**
+** \brief Functions for modifying ts and property
+**
+**
+**/
+
+#include "modifiers/mod_ts_prop.h"
+
+#include "core/rts.h"
+#include "smt-switch/utils.h"
+#include "utils/logger.h"
+#include "utils/term_analysis.h"
+
+using namespace smt;
+using namespace std;
+
+namespace pono {
+
+TransitionSystem pseudo_init_and_prop(TransitionSystem & ts, Term & prop)
```

Comment:
can ts be const?

Reply:
and also prop?

Reply:
Hey, thanks for the review! So `prop` can't because it's edited in-place. And `ts` can't because of this line: https://github.com/upscale-project/pono/blob/c4e299d7b9d10b3f4da9aeec0cf98df86d954f8b/modifiers/mod_ts_prop.cpp#L35

Basically, if it's already a relational TS, we don't copy it. Instead it's just a reference to the original TS. But if you think that's over-optimized, I can just copy it for both cases.